aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -92,13 +93,13 @@ public void seek(long newPos) {
   public int read() throws IOException {
     Preconditions.checkState(!closed, "Cannot read: already closed");
     positionStream();
-
+    int bytesRead = stream.read();
```

Comment:
I'm guessing this is just aligning things with the other `read` method and isn't a strictly necessary change?

Reply:
It's actually also required for correctness because when we retry the input stream we retry from the current position. Previously in this method we advance the position before actually reading the single byte. If we left it as is, upon retry we would read from the position 1 byte ahead of where we needed to re-open the stream from. It's not a problem when there's no retries because if stream.read() fails, it would just fail.